lib/mayaUsd/fileio/translators/translatorMesh.cpp

```diff
@@ -198,11 +198,15 @@ TranslatorMeshRead::TranslatorMeshRead(
     // set mesh name
     const auto& primName = prim.GetName().GetString();
     const auto  shapeName = TfStringPrintf("%sShape", primName.c_str());
-    meshFn.setName(MString(shapeName.c_str()), false, &stat);
 
-    if (!stat) {
-        *status = stat;
-        return;
+    // Set the mesh name if creating a maya mesh object
+    if (transformObj.isNull() || MFn::kMeshData != transformObj.apiType()) {
```

Comment:
Maybe for someone knowing the code it is clear, but to me the comment talking about creating a mesh and the actual tests being done are not correlated, especially since the comment says to set the name of it is a mesh yet it sets the name if the type is **not** mesh-data.

I guess both these tests detect "a mesh object", but it is not obvious.